### PR TITLE
LPS-165649 Fix sidebar on KB template.

### PR DESCRIPTION
**Pull request by Vy-CES**
Hello @ambrinchaudhary 
This P/R is tended to fix the following issue: https://issues.liferay.com/browse/LPS-165649. 

The motivations for the fix are as followed:
1) When the KB templates are selected, we want to append "active" to the class so that the side bar will stay at KB templates URLs.
2) Instead of going to KB edit mode on click at the KB template's title, I want to propose the new behavior at going to KB article view mode when the KB template's title is clicked. 
3) Under development: do you think it is a good idea to at a URL at the top level in KB template's sidebar? When we click into that level it will bring the user to KB template list?

LPS-165649

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
Hey @Vy-CES, thanks for sending this fix to us. Although code LGTM, I have doubts with the behaviour changes. Point 1) is correct, at the view mode we should highligth the current template in the sidebar. 

> 1. Instead of going to KB edit mode on click at the KB template's title, I want to propose the new behavior at going to KB article view mode when the KB template's title is clicked.
> 2. Under development: do you think it is a good idea to at a URL at the top level in KB template's sidebar? When we click into that level it will bring the user to KB template list?

I'm not sure if this is what we want, let me check tomorrow with UX this behaviour. If it is, yes, we should add the Home folder at top level, but actually "home" folder does not exists... and also add a back button. In addittion, it is strange that the sidebar link redirects you to the view of the template, but on the search container list, the link redirects you to the edit.

**Comment by Vy-CES** [comment]
Hey @ambrinchaudhary , thank you for taking your time looking into this. Yes, please check with the UX team to see what is the desired from your team.
My recommendation is that we make KB templates the same way as KB article (without KB folder), it means that we should make it directed to view page on the search container list when we click to the template's title as well.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminNavigationDisplayContext.java [hunk 1]
```diff
@@ -190,7 +190,8 @@ public List<JSONObject> getVerticalNavigationJSONObjects()
 			JSONArray navigationItemsJSONArray = null;
 
 			if (!mvcPath.equals("/admin/view_kb_suggestions.jsp") &&
-				!mvcPath.equals("/admin/view_kb_templates.jsp")) {
+				!mvcPath.equals("/admin/view_kb_templates.jsp") &&
+				!mvcPath.equals("/admin/view_kb_template.jsp")) {
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminNavigationDisplayContext.java:194]: do we really want to mantain both paths? Seems more like a mistake... @adolfopa @AliciaGarciaGarcia ?

**Comment by ambrinchaudhary** [comment]
After revision with UX, there are some changes we would like to ask for this pull: 
1. list of templates in the sidebar should link to the edition, just like the search container list do. Use case for templates is different than for articles. 
2. Template view list must have a back button in the top bar, just like articles have. 
3. Also add a "Home" folder in the sidebar to allow users access to the list of templates. 

Thanks @Vy-CES for taking care of these issues, please send us a new pull and do not hesitate to contact me if you have any doubts.

**Comment by Vy-CES** [comment]
Closed this because I will send this changes on the new P/R

**Pull request closed without merging**